default.nix

```diff
@@ -5,19 +5,19 @@
 self: super:
 
 let
-  src = super.fetchFromGitHub {
-    owner = "ton-blockchain";
-    repo = "ton";
-    rev = "ac3eb1a7b86b4a5351210c4e2670e470f721b7df";
-    sha256 = "1cbv14c60xmy4fanhmf9cdhj845kx6r54b8i0pljkhpv1sy74awd";
+  sources = import ./nix/sources.nix;
+  version = builtins.substring 0 7 sources.ton.rev;
+  src = self.fetchFromGitHub {
+    inherit (sources.ton) owner repo rev;
+    # sadly, not the one niv comes up with. add this check
+    # to make sure people update the hash
+    sha256 = if version == "ceaed40" then
+      "1znikk7l2pv5mdl9rh59dljdrqkbwnazlpdjr4yfc87bcynb1rbz" else
+      "0000000000000000000000000000000000000000000000000000";
```

Comment:
Wait what?

Reply:
`niv` does not supply the correct sha256 hash (fetchSubmodules), so if you run `niv update`, the build will still succeed since you already have the wrong sha256 in cache.